test/watch/test/change-project.test.ts

```diff
@@ -0,0 +1,44 @@
+import { expect, test } from 'vitest'
+import { runInlineTests } from '../../test-utils'
+
+test('reruns tests when config changes', async () => {
+  const { vitest, ctx } = await runInlineTests({
+    'vitest.config.ts': `
+
+    process.stdin.isTTY = true
+    process.stdin.setRawMode = () => process.stdin
+
+    export default {
+      test: {
+        workspace: [
+          './project-1',
+          './project-2',
+        ],
+      },
+    }`,
+    'project-1/vitest.config.ts': { test: { name: 'project-1' } },
+    'project-1/basic-1.test.ts': /* ts */`
+      import { test } from 'vitest'
+      test('basic test 1', () => {})
+    `,
+    'project-2/vitest.config.ts': { test: { name: 'project-2' } },
+    'project-2/basic-2.test.ts': /* ts */`
```

Comment:
Are these intentional? Let's either remove them or use `ts`:

```suggestion
    'project-1/basic-1.test.ts': `
      import { test } from 'vitest'
      test('basic test 1', () => {})
    `,
    'project-2/vitest.config.ts': { test: { name: 'project-2' } },
    'project-2/basic-2.test.ts': `
```

Reply:
These are intentional. They provide syntax highlighting in the IDE without using any runtime functions

Reply:
Technically, `String.raw` can break the code if there is unexpected string interpolation. I think comments like these might be safer

Reply:
@AriPerkkio are you ok with merging this or do you want to have `ts` consistently everywhere? I'm just afraid that at one point we will hit unexpected bug in tests and spend too much time trying to figure out why it happened, so I would advocate for using comments instead